### PR TITLE
Fix automod messages not being parsed/showing up properly

### LINKED ISSUE #2682: (As a Moderator in a Channel)–  'Denying' a Message Caught by AutoMod Shows No Confirmation (though it used to)
#  **Describe the bug:**  #

**Messages that get caught by AutoMod (as a moderator for the channel) don't tell you that they were OR show up at all in the Chatterino chat client when you deny them. There is no confirmation that your 'Deny' action on AutoMod even did anything...**

#  **To reproduce:**  #

**Have a message get caught by AutoMod and be a moderator for the channel in which the message is caught; Deny the AutoMod-caught message.**

#  **Screenshots:**  #

### **The pop-up from AutoMod saying the message was caught, inside of Chatterino, after I clicked 'Deny' and nothing happened, (when something used to)- to verify and confirm that it was in fact denied and not posted to chat, also showing it doesn't let me delete the message caught, either.** ###
_**(I've blocked out usernames to prevent the users involved from harassment of any sort due to their comments.)**_

Chatterino AutoMod Deny Issue 2

### **The completed Mod Action inside of Mod View showing that it did work, and the message caught WAS Denied and not posted to chat.** ###
Chatterino AutoMod Deny Issue 3

#  **Chatterino version:**  #

**Chatterino 2.3.0 (commit 2db140d5a)**

#  **Operating system:**  #

**macOS Big Sur Version 11.2.3**

macOS Big Sur Version 11.2.3

#  **Additional information:**  #

**It used to grey-out the caught message under AutoMod and say a local server message in chat confirming to the moderator/user that it was denied, inside of Chatterino; it no longer does this, and can drive moderators crazy because they don't know if denying it worked or not.**

### PR DESCRIPTION
**Pull request by apa420**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description
Automod didn't work properly but this is now fixed.
Closes #2682

The steps to reproduce the bug are discussed in the issue.

Otherwise:

1. Enable automod to level 4 (for simplicity)
2. Use some account and type something vulgur
3. As a moderator see that the message was held up
4. Click Approve or Deny
5. See that nothing happened to the AutoMod message that was shown when it's supposed to fade to gray

This also fixed automod pubsub messages in general.

Some notes for devs:
When I implemented it 
```
chatterino.pubsub: moderationActionHandlers delete_blocked_term
chatterino.pubsub: moderationActionHandlers delete_permitted_term
chatterino.pubsub: channelTermsActionHandlers add_permitted_term
chatterino.pubsub: channelTermsActionHandlers add_blocked_term
chatterino.pubsub: channelTermsActionHandlers delete_blocked_term
chatterino.pubsub: channelTermsActionHandlers delete_permitted_term
```

Happened while allowing/denying messages and also adding and removing phrases in twitch automod gui.

### CODE REVIEW

File: src/providers/twitch/PubsubClient.cpp [hunk 1]
```diff
@@ -703,16 +726,47 @@ PubSub::PubSub()
...
+            }
+            catch (const std::runtime_error &ex)
+            {
+                qCDebug(chatterinoPubsub)
+                    << "Error parsing channel terms action:" << ex.what();
+            }
         };
 
+    // We don't get this one anymore or anything similiar
+    // We need some new topic so we can listen
```
Comment (iProdigy) [inline src/providers/twitch/PubsubClient.cpp:759]: Just tested `automod-levels-modification.<user-id>.<channel-id>` & it doesn't work with third-party oauth, so we're SOL on this for now

Comment (iProdigy) [inline src/providers/twitch/PubsubClient.cpp:759]: Just tested `automod-levels-modification.<user-id>.<channel-id>` & it *does* work with third-party oauth, but this topic is obviously undocumented (and perhaps overly specific to be worth the subscription)

**Review by iProdigy — APPROVED** [review]
Thanks for fixing this from the mod perspective - looking forward to `automod_message_rejected` & `automod_message_denied` support in the future for the user-side :)

**Comment by AustinHolderr** [comment]
so this would close #2682, right?

**Comment by AustinHolderr** [comment]
@apa420 thank you for your hard work into fixing this issue 💛 i appreciate it

**Review by MrAuro — APPROVED** [review]
works as intended for me 👍

**Review by pajlada — APPROVED** [review]
Looks good to me - good fix for weird problem :ok_hand:

**Pull request merged**